samples/cpp/speech_sample/utils.hpp

```diff
@@ -361,30 +361,103 @@ void sum_performance_counters(std::map<std::string, ov::ProfilingInfo> const& pe
+    std::stringstream ss(s);
+    std::string item;
 
-    if (!str.empty()) {
-        std::string outStr;
-        std::istringstream stream(str);
-        int i = 0;
-        while (getline(stream, outStr, ',')) {
-            auto floatScaleFactor = std::stof(outStr);
-            if (floatScaleFactor <= 0.0f) {
-                throw std::logic_error("Scale factor for input #" + std::to_string(i) +
-                                       " (counting from zero) is out of range (must be positive).");
+    while (getline(ss, item, delim)) {
+        result.push_back(item);
+    }
+    return result;
+}
+
+/**
+ * @brief Concat strings using delimeter
+ * @param chunks input chunks
+ * @param delim delimeter
+ * @return concatenated string
+ */
+std::string concat(const std::vector<std::string>& chunks, char delim) {
+    std::stringstream ss;
+    for (auto&& chunk : chunks) {
+        if (!ss.str().empty()) {
+            ss << delim;
+        }
+        ss << chunk;
+    }
+    return ss.str();
+}
+
+/**
+ * @brief Check whether name is present in node vector
+ * @param nodes nodes
+ * @param node_name name
+ * @return false or true
+ */
+bool check_name(const ov::OutputVector& nodes, const std::string& node_name) {
+    std::vector<std::string> any_names;
+    bool count = false;
+    for (auto& node : nodes) {
+        any_names.push_back(node.get_any_name());
+        auto names = node.get_names();
+        count = std::count(names.begin(), names.end(), node_name);
+        if (count)
+            break;
+    }
+    if (!count) {
+        std::stringstream ss;
+        ss << "Incorrect node name '" + node_name << "'! ";
+        ss << "Try one of the following names: [ ";
+        for (auto&& name : any_names) {
+            ss << name << " ";
+        }
+        ss << "]";
+        throw std::logic_error(ss.str());
+    }
+    return count;
+}
+
+/**
+ * @brief Parse scale factors per input
+ * Format : <input_name1>:<sf1>,<input2>:<sf2> or just <sf>
+ * @param inputs model inputs
+ * @param values_string values_string input string
+ * @return map of scale factors per input
+ */
+std::map<std::string, float> parse_scale_factors(const ov::OutputVector& inputs, const std::string& values_string) {
+    auto check_sf = [&](float sf, const std::string& input_name = "") {
+        if (sf <= 0.0f) {
+            throw std::logic_error("Scale factor for input '" + input_name +
+                                   "' (counting from zero) is out of range (must be positive).");
+        }
+    };
+    std::map<std::string, float> result;
+    auto scale_factor_strings = split(values_string, ',');
+    for (auto& scale_factor_string : scale_factor_strings) {
+        auto values = split(scale_factor_string, ':');
+        if (values.size() == 1) {
+            auto scale_factor = std::stof(values.at(0));
+            check_sf(scale_factor);
+            for (auto& input : inputs) {
+                result[input.get_any_name()] = scale_factor;
```

Comment:
Seems like the resulting map will be overwritten by the last scale_factor.

Reply:
@mryzhov Additional check was added